### PR TITLE
Implement the proof interface

### PR DESCRIPTION
**Pull request by Gilthoniel**
This implements the proof interface for cosipbft and it also uses the chain in the synchronization process.

### CODE REVIEW

**Comment by Gilthoniel** [comment]
The json format cleaning and tests will come in a following PR.

File: core/ordering/cosipbft/blocksync/default_test.go [hunk 1]
```diff
@@ -169,46 +181,71 @@ func TestHandler_Stream(t *testing.T) {
...
 	err = handler.Stream(fake.Sender{}, fake.NewReceiver(msgs...))
 	require.Error(t, err)
 	require.Regexp(t, "pbft catch up failed: mismatch link '[0]{8}' != '[0-9a-f]{8}'", err.Error())
 
-	err = handler.Stream(fake.NewBadSender(), fake.NewReceiver(types.NewSyncMessage(0)))
+	err = handler.Stream(fake.NewBadSender(), fake.NewReceiver(types.NewSyncMessage(makeChain(t, 0))))
 	require.EqualError(t, err, "sending ack failed: fake error")
 }
 
 // Utility functions -----------------------------------------------------------
```
Comment (nkcr) [inline core/ordering/cosipbft/blocksync/default_test.go:209]: I don't mind changing to this, but then we should stay consistent with the rest, as we normally use

```
// -----------------------------------------------------------------------------
// Utility functions
```

Comment (Gilthoniel) [reply core/ordering/cosipbft/blocksync/default_test.go:209]: I replaced the occurrences.

Comment (nkcr) [reply core/ordering/cosipbft/blocksync/default_test.go]: you put the line at the bottom, instead of top 😅

Comment (Gilthoniel) [reply core/ordering/cosipbft/blocksync/default_test.go]: oopsy !

**Review by nkcr — APPROVED** [review]

**Pull request merged**